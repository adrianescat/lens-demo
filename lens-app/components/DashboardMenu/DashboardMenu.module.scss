.aside {
  display: flex;
  flex-direction: row;
  width: 100%;
  border-bottom: 1px solid #f0f0f0;

  ul {
    width: 100%;
    display: flex;
    justify-content: space-evenly;
    align-items: center;
  }

  li {
    font-size: 20px;
    height: 50px;
    display: flex;
    align-items: center;
    cursor: pointer;

    &:hover {
      color: #03556f;
    }

    &.disabled,
    &.disabled:hover {
      cursor: not-allowed;
      color: grey;
    }
  }
}

@media screen and (min-width: 930px) {
  .aside {
    flex-direction: row;
    width: 20%;
    border-bottom: none;
    border-right: 1px solid #f0f0f0;
    height: 89vh;

    ul {
    width: 100%;
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    flex-direction: column;
  }
  }
}