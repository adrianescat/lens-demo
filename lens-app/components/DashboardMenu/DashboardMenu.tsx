import React, { ReactElement } from 'react'
import styles from './DashboardMenu.module.scss';
import { DashboardCategories } from '../Dashboard/Dashboard';

export interface Props {
  setCategory: (categoryName: DashboardCategories) => void;
}

export default function DashboardMenu({ setCategory }: Props): ReactElement {
  return (
    <aside className={styles.aside}>
      <ul>
        <li onClick={() => setCategory(DashboardCategories.LENSLIST)}>
          Lens
          </li>
        <li onClick={() => setCategory(DashboardCategories.PATIENTS)}>
          Patients
          </li>
        <li onClick={() => setCategory(DashboardCategories.HOSPITALS)}>
          Hospitals
          </li>
      </ul>
    </aside>
  )
}
