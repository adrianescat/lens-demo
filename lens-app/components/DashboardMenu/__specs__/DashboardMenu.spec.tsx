import React from 'react';
import { shallow } from 'enzyme';
import DashboardMenu, { Props } from '../DashboardMenu';

let wrapper: any;
let props: Props;

beforeEach(() => {
  props = { setCategory: jest.fn() }
  wrapper = shallow(<DashboardMenu {...props} />);
});

test('should match snapshots', () => {
  expect(wrapper).toMatchSnapshot();
});