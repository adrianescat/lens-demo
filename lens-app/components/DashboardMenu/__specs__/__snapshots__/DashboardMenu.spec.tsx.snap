// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<aside
  className="aside"
>
  <ul>
    <li
      onClick={[Function]}
    >
      Lens
    </li>
    <li
      onClick={[Function]}
    >
      Patients
    </li>
    <li
      onClick={[Function]}
    >
      Hospitals
    </li>
  </ul>
</aside>
`;
